src/main/kotlin/com/performance/domain/reservation/service/ReservationService.kt

```diff
@@ -0,0 +1,31 @@
+package com.performance.domain.reservation.service
+
+import com.performance.domain.performance.repository.PerformanceRepository
+import com.performance.domain.reservation.dto.ReservationRequest
+import com.performance.domain.reservation.model.Reservation
+import com.performance.domain.reservation.repository.ReservationRepository
+import com.performance.domain.seat.repository.SeatRepository
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class ReservationService(
+    private val reservationRepository: ReservationRepository,
+    private val performanceRepository: PerformanceRepository,
+    private val seatRepository: SeatRepository
+) {
+    @Transactional
+    fun saveReservation(request: ReservationRequest) =
+        reservationRepository.save(
+            Reservation(
+                id = request.id,
+                email = request.email,
+                performance = performanceRepository.findByIdOrNull(request.performanceId)
+                    ?: throw IllegalStateException(),
+                seat = seatRepository.findByIdOrNull(request.seatId)
+                    ?: throw IllegalStateException(),
+                status = request.status
+            )
+        )
+}
```

Comment:
```suggestion
    @Transactional
    fun saveReservation(request: ReservationRequest): Reservation {
        val performance = performanceRepository.findByIdOrNull(request.performanceId) ?: throw IllegalStateException()
        val seat = seatRepository.findByIdOrNull(request.seatId) ?: throw IllegalStateException()

        return reservationRepository.save(
            Reservation(
                id = request.id,
                email = request.email,
                performance = performance,
                seat = seat,
                status = request.status
            )
        )
    }
```

kotlin으로 개발좀 하다 보니, 변수 선언하는게 가독성 좋더라구요 ㅎ.,,,.

Reply:
변수 선언을 하는 게 좀 터 코틀린다운 건가요?

Reply:
코틀린다운건 모르겠지만, 가변 데이터가 아니라면 데이터 선언해주는게 가독성도 좋고 로직 짜기에도 편하더라구요.